docs/design_guide.md

```diff
@@ -0,0 +1,285 @@
+```python
+"""One line function description.
+
+Longer explanation...
+...
+
+Args
+    arg1 (int): I am an integer argument.
+    arg2 (str): I am a string argument.
+    arg3: I am argument without a type specified.
+
+Returns:
+    bool: This function returns a boolean.
+
+Raises:
+    TypeError: explanation of error.
+"""
+```
+
+### Naming
+
+| Name Type | Convention | Notes |
+| :---      | :---       | :---  |
+|Package, Module | `shortlowcase` | Underscores can be used for readability |
+|Class | `CapWords` | |
+| Type Variable | `CapWords_suffix`| |
+| Global Variables | `low_case_sep`| |
+| Functions | `low_case_sep`| |
+| Function Variables | `low_case_sep`| |
+| Function Arguments | `low_case_sep` | Use underscores sparingly |
+| Constants | `ALL_CAPS` | |
+
+## Development Environment
+
+The library will be written in Python 3.7, so use this version when developing. To see what version you currently have installed, run `python3 --version`. If the output is `Python 3.7.x`, you are good to go! Otherwise, [install](https://www.python.org/downloads/).
+
+Code can be linted using [flake8](https://flake8.pycqa.org/en/latest/index.html), just ensure it is installed for the correct version of Python (i.e. 3.7).
+
+To make standard formatting easy and consistent, the auto formatting tool [yapf](https://github.com/google/yapf) should be used.
+
+*Note:* `flake8` and `yapf` will not ensure that all of the standards for this library are met&mdash;It is your responsibility to do so.
+
+Use any editor you want! Simply ensure that your editor will not interfere with any of the standards outlined above. If linting is enabled, ensure the linter is `flake8` and if possible, modify its configuration to match the one presented in the Makefile. Some editors can integrate `yapf` as a plugin&mdash;this is highly recommended.
+
+## Library Documentation
+
+Current documentation [here](https://r2b2.readthedocs.io/en/latest/)
+
+[Sphinx](https://www.sphinx-doc.org/en/master/) and [ReadTheDocs](https://docs.readthedocs.io/en/latest/index.html) will be used to generate well-formatted documentation for the library.
+
+**TODO** add to section.
+
+## Testing
+
+We will [pytest](https://docs.pytest.org/en/latest/) for testing source code and [tox](https://tox.readthedocs.io/en/latest/) to provide a standard testing environment.
+
+**TODO** add to this section.
+
+## Basic Workflow Notes
+
+**Repository**
+
+- Develop in 1 repo only, preferably a new repo owned by a GitHub organization (rather than a personal account)
+- **Branches** are for individual features or project sections
+  - Preferably, multiple people will not be working heavily on the same branch at the same time
+  - If you get carried away and realize you are working on functionality that does not belong on the current branch, **stash** it for later.
+
+
+**Pushing, Merging, and Pulling**
+
+- **Commit** messages should have a short one-line summary, followed by a detailed description of changes made
```

Comment:
> **Commit** messages should have a one-line summary of 50 characters or less.

Reply:
Are you suggesting commit messages limited to 50 characters and no more? The first line should be short, but for larger commits, there should be a detailed description of changes after this one-line summary.

Reply:
Yes - I agree. Sorry for the confusion. I was just proposing a modification to the first clause up to the comma.